src/go/k8s/controllers/redpanda/cluster_controller.go

```diff
@@ -113,155 +116,85 @@ func (r *ClusterReconciler) Reconcile(
 	log.Info("Starting reconcile loop")
 	defer log.Info("Finished reconcile loop")
 
-	var redpandaCluster vectorizedv1alpha1.Cluster
-	crb := resources.NewClusterRoleBinding(r.Client, &redpandaCluster, r.Scheme, log)
-	if err := r.Get(ctx, req.NamespacedName, &redpandaCluster); err != nil {
+	var vectorizedCluster vectorizedv1alpha1.Cluster
+	ar := newAttachedResources(ctx, r, log, &vectorizedCluster)
+	if err := r.Get(ctx, req.NamespacedName, &vectorizedCluster); err != nil {
 		// we'll ignore not-found errors, since they can't be fixed by an immediate
 		// requeue (we'll need to wait for a new notification), and we can get them
 		// on deleted requests.
 		if apierrors.IsNotFound(err) {
-			if removeError := crb.RemoveSubject(ctx, req.NamespacedName); removeError != nil {
+			if removeError := ar.getClusterRoleBinding().RemoveSubject(ctx, req.NamespacedName); removeError != nil {
 				return ctrl.Result{}, fmt.Errorf("unable to remove subject in ClusterroleBinding: %w", removeError)
 			}
 			return ctrl.Result{}, nil
 		}
 		return ctrl.Result{}, fmt.Errorf("unable to retrieve Cluster resource: %w", err)
 	}
-	// if the cluster is being deleted, delete finalizers
-	if !redpandaCluster.GetDeletionTimestamp().IsZero() {
-		return r.handleClusterDeletion(ctx, &redpandaCluster, log)
+
+	r.isManagedByThisController = isRedpandaClusterManaged(log, &vectorizedCluster) &&
+		isRedpandaClusterVersionManaged(log, &vectorizedCluster, r.RestrictToRedpandaVersion)
```

Comment:
Why property that is true for one Cluster custom resource should be used by another? I think it should be function scoped not struct scoped.

Reply:
Hmm, true. I was thinking I would check it in other methods. Right now it's only checked for deletion and nowhere else. This is a bug.

Reply:
It's ok to have this reconciler scoped since we set it first thing in the reconcile. Only one reconcile will be done at a time. This is a controller-runtime guarantee.

Reply:
Oh, no... it's not a bug. If it's not being managed, we remove the finalizer and don't manage it. :boom: